lib/ansible/modules/user.py

```diff
@@ -801,16 +820,23 @@ def modify_user_usermod(self):
 
             if self.expires < time.gmtime(0):
                 if current_expires >= 0:
-                    cmd.append('-e')
-                    cmd.append('')
+                    if self.local:
+                        lexpires = -1
+                    else:
+                        cmd.append('-e')
+                        cmd.append('')
             else:
                 # Convert days since Epoch to seconds since Epoch as struct_time
                 current_expire_date = time.gmtime(current_expires * 86400)
 
                 # Current expires is negative or we compare year, month, and day only
                 if current_expires < 0 or current_expire_date[:3] != self.expires[:3]:
-                    cmd.append('-e')
-                    cmd.append(time.strftime(self.DATE_FORMAT, self.expires))
+                    if self.local:
+                        # Convert seconds since Epoch to days since Epoch
+                        lexpires = int(math.floor(float(self.module.params['expires']))) // 86400
```

Comment:
Same here: I don't believe the inner `float()` conversion is necessary.

Reply:
I guess you are correct. Will change it and let us see what the testing says.

Reply:
Done in https://github.com/ansible/ansible/pull/72022/commits/afd6e08b4fa3ef2e002f67dfc7e0efb6fbcc15f2